scilpy/tracking/tools.py

```diff
@@ -147,28 +149,187 @@ def get_subset_streamlines(sft, max_streamlines, rng_seed=None):
         Maximum number of streamlines to output.
     rng_seed: int
         Random number to use for shuffling the data.
+    return_indices_only: bool
+        If true, return a random list of indices. Else, return the Stateful
+        Tractogram containing the chosen streamlines.
 
     Return
     ------
-    subset_sft: StatefulTractogram
-        The filtered streamlines as a sft.
+    subset_sft: StatefulTractogram or list
+        The filtered streamlines as a sft, or the list of indices if
+        return_indices_only.
     """
 
     rng = np.random.RandomState(rng_seed)
     ind = np.arange(len(sft.streamlines))
     rng.shuffle(ind)
 
-    subset_streamlines = list(np.asarray(sft.streamlines, dtype=object)[
-                              ind[:max_streamlines]])
-    subset_data_per_point = sft.data_per_point[ind[:max_streamlines]]
-    subset_data_per_streamline = sft.data_per_streamline[ind[:max_streamlines]]
+    if return_indices_only:
+        return ind[:max_streamlines]
 
-    subset_sft = StatefulTractogram.from_sft(
-        subset_streamlines, sft,
-        data_per_point=subset_data_per_point,
-        data_per_streamline=subset_data_per_streamline)
+    return sft[ind[:max_streamlines]]
 
-    return subset_sft
+
+def get_n_subsets_streamlines_sequentially(orig_sft, chunk_sizes):
+    """
+    Get n chunks of subsets of streamlines, sequentially through the sft.
+
+    Parameters
+    ----------
+    orig_sft: StatefulTractogram
+        Initial tractogram to subdivide
+    chunk_sizes: list[int]
+        Number of streamlines to keep per chunk.
+
+    Return
+    ------
+    all_chunks: list[StatefulTractogram]
+        The filtered streamlines as a sfts.
+    """
+    if sum(chunk_sizes) > len(orig_sft):
+        raise ValueError("You asked for more streamlines than are available.")
+
+    nb_chunks = len(chunk_sizes)
+
+    curr = 0
+    sfts = []
+    for i in range(nb_chunks):
+        nb_str = chunk_sizes[i]
+        sfts.append(orig_sft[curr:curr + nb_str])
+        curr += chunk_sizes[i]
+
+    return sfts
+
+
+def get_n_subsets_streamlines_random(orig_sft, chunk_sizes, seed,
+                                     return_indices_only=False):
+    """
+    Get n chunks of subsets of streamlines.
```

Comment:
`n` is `len(chunk_sizes)`? Maybe we should think of a better name if that's the case? Like `random_split_streamlines` or something? Also isn't `get_subset_streamlines` the `n=1` case? Could we merge both methods?

Reply:
> n is len(chunk_sizes)? Maybe we should think of a better name if that's the case? Like random_split_streamlines or something?

I tried updating the doc and names. Tell me if it's clearer.

> Also isn't get_subset_streamlines the n=1 case? Could we merge both methods?

Yes it is; `get_n_subsets _random` called `get_subset_streamlines` n times. But also, it is used in scil_resample_tractogram, so I could merge it but it would return a list of one subset instead of only the subset, and I should change scil_resample_tractogram to get the first result. Do you want me to do that?

Reply:
Maybe we should get a second opinion before you start refactoring ? @arnaudbore @frheault